core/network/endpoint_bindings_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network_test
+
+import (
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/network"
+)
+
+var _ = gc.Suite(&bindingsMockSuite{})
+
+type bindingsMockSuite struct {
+	testing.IsolationSuite
+}
+
+func (s *bindingsMockSuite) TestMapBindingsWithSpaceNames(c *gc.C) {
+	infos := s.expectedSpaceInfos()
+
+	initial := map[string]string{
+		"db":      "2",
+		"testing": "3",
+		"empty":   network.AlphaSpaceId,
+	}
+
+	withSpaceNames, err := network.MapBindingsWithSpaceNames(initial, infos)
+	c.Assert(err, jc.ErrorIsNil)
+
+	expected := map[string]string{
+		"db":      "two",
+		"testing": "three",
+		"empty":   network.AlphaSpaceName,
+	}
+	c.Assert(withSpaceNames, jc.DeepEquals, expected)
+}
+
+func (s *bindingsMockSuite) TestMapBindingsWithSpaceNamesWithNoLookup(c *gc.C) {
+	initial := map[string]string{
+		"db":      "2",
+		"testing": "3",
+		"empty":   network.AlphaSpaceId,
+	}
+
+	_, err := network.MapBindingsWithSpaceNames(initial, nil)
+	c.Assert(err, gc.ErrorMatches, "*not valid*")
```

Comment:
Can we not do error asserts on strings. This is not what forms the contract with the caller.

Consider `c.Check(err, jc.ErrorIs, coreerrors.NotValid)`